src/scripts/Player.ts

```diff
@@ -18,6 +18,8 @@ class Player {
     private _shinyList: boolean[];
     private _notCaughtBallSelection: GameConstants.Pokeball;
     private _alreadyCaughtBallSelection: GameConstants.Pokeball;
+    private _sortOption: KnockoutObservable<string>;
```

Comment:
The advantage of using enumerators is that you can make sure you only one of the values that you want to return, and you have those values in a nice concise place. For example with the GameStates you can easily see a list of all the possible states.

In this case you could make 
```
    export enum SortOption{
        "Shiny",
        "Level",
        "Attack",
    }
```

Then giving `_sortDirection` the type `KnockoutObservable<GameConstants.SortOption>`
Since the enum is a string, the conversion between `value` and the variable should work.